fxa-auth-db-server/test/backend/db_tests.js

```diff
@@ -299,16 +314,144 @@ module.exports = function(config, DB) {
+                  uaOS: 'c',
+                  uaOSVersion: 'd',
+                  uaDeviceType: 'e'
+                })
+              })
+              .then(function (result) {
+                t.deepEqual(result, {}, 'Returned an empty object on session token creation')
+
+                // Fetch all of the sessions tokens for the account
+                return db.sessions(ACCOUNT.uid)
+              })
+              .then(function (sessions) {
+                t.equal(sessions.length, 2, 'sessions contains one item')
+                var index = 0
+                if (sessions[0].tokenId.toString('hex') === SESSION_TOKEN_ID.toString('hex')) {
+                  index = 1
+                }
+                t.equal(sessions[index].tokenId.toString('hex'), VERIFIED_SESSION_TOKEN_ID.toString('hex'), 'tokenId is correct')
+                t.equal(sessions[index].uid.toString('hex'), ACCOUNT.uid.toString('hex'), 'uid is correct')
+                t.equal(sessions[index].uaBrowser, 'a', 'uaBrowser is correct')
+                t.equal(sessions[index].uaBrowserVersion, 'b', 'uaBrowserVersion is correct')
+                t.equal(sessions[index].uaOS, 'c', 'uaOS is correct')
+                t.equal(sessions[index].uaOSVersion, 'd', 'uaOSVersion is correct')
+                t.equal(sessions[index].uaDeviceType, 'e', 'uaDeviceType is correct')
+
+                // Fetch the verified session token
+                return db.sessionToken(VERIFIED_SESSION_TOKEN_ID)
+              })
+              .then(function(token) {
+                t.deepEqual(token.uid, ACCOUNT.uid, 'token belongs to this account')
+                t.equal(token.uaBrowser, 'a', 'uaBrowser is correct')
+                t.equal(token.uaBrowserVersion, 'b', 'uaBrowserVersion is correct')
+                t.equal(token.uaOS, 'c', 'uaOS is correct')
+                t.equal(token.uaOSVersion, 'd', 'uaOSVersion is correct')
+                t.equal(token.uaDeviceType, 'e', 'uaDeviceType is correct')
+                t.equal(!!token.emailVerified, ACCOUNT.emailVerified, 'token emailVerified is same as account emailVerified')
+                t.equal(token.tokenVerificationId, undefined, 'tokenVerificationId is undefined')
+
+                // Fetch the verified session token with its verification state
+                return db.sessionTokenWithVerificationStatus(VERIFIED_SESSION_TOKEN_ID)
+              })
+              .then(function (token) {
+                t.equal(token.tokenVerificationId, null, 'tokenVerificationId is null')
+
+                // Attempt to verify session token with invalid tokenVerificationId
+                return db.verifyToken(hex16(), { uid: ACCOUNT.uid })
+              })
+              .then(function () {
+                t.fail('Verifying session token with invalid tokenVerificationId should have failed')
+              }, function () {
+                t.pass('Verifying session token with invalid tokenVerificationId failed as expected')
```

Comment:
I think it might be useful to add some extra asserts for the error code/message here. Which I believe should be `errno:116`